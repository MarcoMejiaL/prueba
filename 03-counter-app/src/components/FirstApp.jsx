import Title from '../add/Title'
import { Button } from '../add/Button'
const nombre ={
    nombre:'Marco Antonio',
    apellido:'Mejia Lopez',
    number:32
}



export const FirstApp=()=>{
   
    return( 
        <>{/* fragmento */}
        
            <Title
            name={nombre.nombre}
            firstNamr={nombre.apellido}
            number={nombre.number}
            />
           <Button></Button>
            
        
        
   
    <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Minus, aspernatur! Ipsa eligendi ratione culpa voluptatibus aperiam repudiandae quasi placeat in porro possimus modi beatae dolores quisquam tempore molestias, tempora doloremque!</p>
    </>
    )
}

