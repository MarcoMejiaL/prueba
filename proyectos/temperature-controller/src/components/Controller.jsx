import { useState } from "react"

 const Controller=()=>{
    const [temperature,setTemperature]=useState(10)
    const [temperatureType, setTemperatureType]=useState('C')

    const  handelAdd=()=>{
        setTemperature(temperature + 1)

    }
    const handelMinus=()=>{
        setTemperature(temperature - 1)
        
    }
    const convertidor=()=>{
        if(temperatureType==='C'){
            setTemperatureType('F')
            setTemperature((temperature * (9/5) )+32)
        }
        else{
            setTemperatureType('C')
            setTemperature(((temperature-32)*(5/9)))
        }
    }
    return(
        <section className="container">
            <div className="forma">
                <div className="title__container">
                <h2>{temperature.toFixed(2)}Â°{temperatureType}</h2>
                </div>
            
            <div className="button__container">
            <button onClick={handelAdd} className="main__button">+</button>
            <button onClick={handelMinus } className="main__button">-</button>

            </div>
            <div className="button__container--secondary">
                <button className="secondary__button" onClick={convertidor}>{temperatureType}</button>
            </div>
            </div>
        </section>
    )
}
export default Controller;