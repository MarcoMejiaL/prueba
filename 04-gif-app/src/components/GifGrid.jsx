

import { GifItem } from "../components/GifItem"; 
 
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category)
  
  /* 
  const [images, setimages] = useState([]);


  const getImages = async()=>{
    const newImages = await getGif(category);
    setimages(newImages);
  }

useEffect(() => {
  getImages();

  
}, [])
 */

  return (
    <>
        <h3>{category}</h3>
        {
          isLoading && (<h2>Cargando.....</h2>)
        }

       <div className="card-grid">
        {
          images.map((images)=>(
            <GifItem 
            key={images.id}
            {...images}
            
            />
            
          ))
        }
        </div> 
    </>
  )
}
