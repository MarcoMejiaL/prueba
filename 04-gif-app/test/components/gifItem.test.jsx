
import { render,screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('testeo de gifItem',()=>{
    
    const url = 'https://one-punch.com/saitama.jpg'
    const title = 'Saitama'
    test('gif item tiene que tener un valor',()=>{
        
        const {container} =render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();

        
        
    });
    test('debe de mostrar la imagen con el URL y el ALT indicado',()=>{
        render(<GifItem title={title} url={url}/>);
        /* screen.debug(); */
        const {src, alt} = screen.getByRole('img');
       /*  expect(screen.getByRole('img').src).toBe(url)
        */
        expect(src).toBe(url)
        expect(alt).toBe(title )

        
        
    })
    test('se tiene que mostrar el titulo',()=>{
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})