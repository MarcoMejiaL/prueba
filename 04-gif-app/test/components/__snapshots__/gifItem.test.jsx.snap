// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testeo de gifItem gif item tiene que tener un valor 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
